cpp/src/parquet/column_reader-test.cc

```diff
@@ -102,7 +102,7 @@ class TestPrimitiveReader : public ::testing::Test {
           &vresult[0] + total_values_read, &values_read));
       total_values_read += static_cast<int>(values_read);
       batch_actual += batch;
-      batch_size = std::max(batch_size * 2, 4096);
+      batch_size = std::min(1 << 24, std::max(batch_size * 2, 4096));
```

Comment:
Worth adding `kMinimumBatchSize` `kMaximumBatchSize`, bonus point for `arrow::clamp`.